<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yuxin Zou" />


<title>SuSiE RSS Diagnostic</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">susierss_diagnostic</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">SuSiE RSS Diagnostic</h1>
<h4 class="author">Yuxin Zou</h4>
<h4 class="date">1/19/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>susierss_diagnostic/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201222code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201222)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201222code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201222)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomzouyuxinsusierssdiagnostictree26f87300a7b99b89124cb6b5a59a9f82553e1705targetblank26f8730a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/zouyuxin/susierss_diagnostic/tree/26f87300a7b99b89124cb6b5a59a9f82553e1705" target="_blank">26f8730</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomzouyuxinsusierssdiagnostictree26f87300a7b99b89124cb6b5a59a9f82553e1705targetblank26f8730a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/zouyuxin/susierss_diagnostic/tree/26f87300a7b99b89124cb6b5a59a9f82553e1705" target="_blank">26f8730</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  data/susierss_diagnostic_query.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/diagnostic.Rmd</code>) and HTML (<code>docs/diagnostic.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zouyuxin/susierss_diagnostic/14759e07e9237f0a405ea6e492fbb0f65a40b090/docs/diagnostic.html" target="_blank">14759e0</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/bc8c722ce4e760c751d9ace9bc64913bdefecee5/analysis/diagnostic.Rmd" target="_blank">bc8c722</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-30
</td>
<td>
wflow_publish(“analysis/diagnostic.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/zouyuxin/susierss_diagnostic/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/diagnostic.html" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/d454fcd8f6d7d37b209b7c4a4c88fde224fcd442/analysis/diagnostic.Rmd" target="_blank">d454fcd</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
<td>
wflow_publish(“analysis/diagnostic.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>In diagnostic, we assume the following model <span class="math display">\[
\hat{z} \sim N(Rz, \sigma^2 R + s^2 I), \sigma^2 + s^2 \leq 1.
\]</span> Let <span class="math inline">\(R = U D U^\intercal\)</span> be eigen-decomposition of <span class="math inline">\(R\)</span>, <span class="math inline">\(D\)</span> has <span class="math inline">\(p_1\)</span> positive diagonal elements, <span class="math inline">\(U = [U_1 \ U_2]\)</span>, <span class="math inline">\(U_1\)</span> is a <span class="math inline">\(p\)</span> by <span class="math inline">\(p_1\)</span> matrix, <span class="math inline">\(U_2\)</span> is a <span class="math inline">\(p\)</span> by <span class="math inline">\(p_2=p-p_1\)</span> matrix. We transform <span class="math inline">\(\hat{z}\)</span> into null space of <span class="math inline">\(R\)</span> by multiplying <span class="math inline">\(U_2^\intercal\)</span> <span class="math display">\[
U_2^T \hat{z} \sim N(0, s^2 I_{p_2}).
\]</span> Therefore, <span class="math inline">\(s^2\)</span> can be estimated as <span class="math inline">\(\frac{1}{p_2} \hat{z}^\intercal U_2 U_2^\intercal \hat{z}\)</span>. We estimate <span class="math inline">\(\sigma^2\)</span> by <span class="math inline">\(1-s^2\)</span>.</p>
<p>Under the null, the posterior distribution of <span class="math inline">\(\hat{z}_j | \hat{z}_{-j}\)</span> is: <span class="math display">\[
\hat{z}_j | \hat{z}_{-j} \sim N(-\frac{1}{\Omega_{jj}} \Omega_{j,-j} \hat{z}_{-j},1/\Omega_{jj}), \Omega = (\sigma^2 R + s^2 I)^{-1}.
\]</span></p>
<p>The estimated <span class="math inline">\(s^2\)</span> could be very small, which leads to small variance in the posterior distribution. In the simulation, we tried truncated method, <span class="math inline">\(s^2 = max(0.005, s^2)\)</span>. We also tried to estimate <span class="math inline">\(s^2\)</span> from pseudo likelihood using conditional distributions.</p>
<p>In the following simulation, we extract 200 regions from 1000 Genome data using GBR (British in England and Scotland), IBS (Iberian Populations in Spain) and CLM (Colombians in Medellin, Colombia) population. We got 1000 Genome genotypes from <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/supporting/hd_genotype_chip" class="uri">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/supporting/hd_genotype_chip</a>, ALL.chip.omni_broad_sanger_combined.20140818.snps.genotypes.vcf.gz. The VCF file is converted to bed format using plink. We removed related samples. There are 103 GBR samples, 150 IBS samples and 107 CLM samples. We generated z scores based on GBR samples and we want to detect the inconsistency between z scores and LD (from GBR, IBS or CLM).</p>
<p>In summary, a large <span class="math inline">\(s^2\)</span> indicates that the z scores and LD matrix are mismatched. One mismatch is that the LD matrix is from the reference panel. The <span class="math inline">\(s^2\)</span> using LD from CLM is larger than the one using LD from IBS. The other mismatch is that some SNPs have flipped z scores.</p>
<pre class="r"><code>library(tibble)
library(dplyr)
out = readRDS(&#39;data/susierss_diagnostic_query.rds&#39;)
out = as_tibble(out)
out_null = out %&gt;% filter(simulate.n_signal == 0, flip_z.flip == FALSE)
out_signal = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == FALSE)
out_null_flip = out %&gt;% filter(simulate.n_signal == 0, 
                               flip_z.flip == TRUE, 
                               flip_z.flip_pos == &#39;null&#39;)
out_signal_flip_null = out %&gt;% filter(simulate.n_signal == 1, 
                               flip_z.flip == TRUE, 
                               flip_z.flip_pos == &#39;null&#39;)
out_signal_flip_signal = out %&gt;% filter(simulate.n_signal == 1, 
                               flip_z.flip == TRUE, 
                               flip_z.flip_pos == &#39;signal&#39;)</code></pre>
<pre class="r"><code>out_null_sample = out_null %&gt;% filter(susie_rss.ld_type == &#39;sample&#39;, susie_rss.correct_ld == FALSE)</code></pre>
<p>Here is a summary of number of SNPs in each region:</p>
<pre class="r"><code>summary(sapply(out_null_sample$diagnostic.res1_simple, function(x) nrow(x$post)))</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  191.0   425.0   497.5   519.8   598.0  1016.0 </code></pre>
<div id="simulation-under-null-z-0" class="section level2">
<h2>Simulation under null (z = 0)</h2>
<pre class="r"><code>out_null_sample = out_null %&gt;% filter(susie_rss.ld_type == &#39;sample&#39;, susie_rss.correct_ld == FALSE)
out_null_IBS = out_null %&gt;% filter(susie_rss.ld_type == &#39;IBS&#39;, susie_rss.correct_ld == FALSE)
out_null_CLM = out_null %&gt;% filter(susie_rss.ld_type == &#39;CLM&#39;, susie_rss.correct_ld == FALSE)
out_null_IBSz = out_null %&gt;% filter(susie_rss.ld_type == &#39;IBS&#39;, susie_rss.correct_ld == TRUE)
out_null_CLMz = out_null %&gt;% filter(susie_rss.ld_type == &#39;CLM&#39;, susie_rss.correct_ld == TRUE)</code></pre>
<div id="check-s2" class="section level3">
<h3>Check <span class="math inline">\(s^2\)</span></h3>
<pre class="r"><code>par(mfrow=c(2,3))
out_null_sample_s2 = apply(out_null_sample, 1, function(x) x$diagnostic.res1_simple$s2)
out_null_sample_s2_pseudo = apply(out_null_sample, 1, function(x) x$diagnostic.res3$s2)
out_null_IBS_s2 = apply(out_null_IBS, 1, function(x) x$diagnostic.res1_simple$s2)
out_null_IBS_s2_pseudo = apply(out_null_IBS, 1, function(x) x$diagnostic.res3$s2)
out_null_CLM_s2 = apply(out_null_CLM, 1, function(x) x$diagnostic.res1_simple$s2)
out_null_CLM_s2_pseudo = apply(out_null_CLM, 1, function(x) x$diagnostic.res3$s2)

# out_null_IBSz_s2 = apply(out_null_IBSz, 1, function(x) x$diagnostic.res1_simple$s2)
# out_null_CLMz_s2 = apply(out_null_CLMz, 1, function(x) x$diagnostic.res1_simple$s2)

plot(out_null_sample_s2, out_null_IBS_s2, xlab = &#39;LD = GBR&#39;, ylab=&#39;LD = IBS&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)
plot(out_null_sample_s2, out_null_CLM_s2, xlab = &#39;LD = GBR&#39;, ylab=&#39;LD = CLM&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)
plot(out_null_IBS_s2, out_null_CLM_s2, xlab = &#39;LD = IBS&#39;, ylab=&#39;LD = CLM&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)

# plot(out_null_sample_s2, out_null_IBSz_s2, xlab = &#39;LD = GBR&#39;, ylab=&#39;LD = IBS z&#39;, main=&#39;GBR z scores&#39;)
# abline(0,1)
# plot(out_null_sample_s2, out_null_CLMz_s2, xlab = &#39;LD = GBR&#39;, ylab=&#39;LD = CLM z&#39;, main=&#39;GBR z scores&#39;)
# abline(0,1)

plot(out_null_sample_s2, out_null_sample_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;pseudo likelihood&#39;, main=&#39;GBR&#39;)
abline(0,1)
plot(out_null_IBS_s2, out_null_IBS_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;pseudo likelihood&#39;, main=&#39;IBS&#39;)
abline(0,1)
plot(out_null_CLM_s2, out_null_CLM_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;pseudo likelihood&#39;, main=&#39;CLM&#39;)
abline(0,1)</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-5-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The <span class="math inline">\(s^2\)</span> from pseudo likelihood is similar to the standard MLE estimates.</p>
</div>
<div id="check-conditional-mean" class="section level3">
<h3>Check conditional mean</h3>
<pre class="r"><code>z = unlist(lapply(out_null_sample$diagnostic.res1_simple, function(x) x$post$z))
out_null_sample_mean = unlist(lapply(out_null_sample$diagnostic.res1_simple, function(x) x$post$postmean))
out_null_sample_z = unlist(lapply(out_null_sample$diagnostic.res1_simple, function(x) x$post$post_z))
out_null_IBS_mean = unlist(lapply(out_null_IBS$diagnostic.res1_simple, function(x) x$post$postmean))
out_null_IBS_z = unlist(lapply(out_null_IBS$diagnostic.res1_simple, function(x) x$post$post_z))
out_null_CLM_mean = unlist(lapply(out_null_CLM$diagnostic.res1_simple, function(x) x$post$postmean))
out_null_CLM_z = unlist(lapply(out_null_CLM$diagnostic.res1_simple, function(x) x$post$post_z))

out_null_sample_trun_mean = unlist(lapply(out_null_sample$diagnostic.res2, function(x) x$post$postmean))
out_null_sample_trun_z = unlist(lapply(out_null_sample$diagnostic.res2, function(x) x$post$post_z))
out_null_IBS_trun_mean = unlist(lapply(out_null_IBS$diagnostic.res2, function(x) x$post$postmean))
out_null_IBS_trun_z = unlist(lapply(out_null_IBS$diagnostic.res2, function(x) x$post$post_z))
out_null_CLM_trun_mean = unlist(lapply(out_null_CLM$diagnostic.res2, function(x) x$post$postmean))
out_null_CLM_trun_z = unlist(lapply(out_null_CLM$diagnostic.res2, function(x) x$post$post_z))

out_null_sample_pseudo_mean = unlist(lapply(out_null_sample$diagnostic.res3, function(x) x$post$postmean))
out_null_sample_pseudo_z = unlist(lapply(out_null_sample$diagnostic.res3, function(x) x$post$post_z))
out_null_IBS_pseudo_mean = unlist(lapply(out_null_IBS$diagnostic.res3, function(x) x$post$postmean))
out_null_IBS_pseudo_z = unlist(lapply(out_null_IBS$diagnostic.res3, function(x) x$post$post_z))
out_null_CLM_pseudo_mean = unlist(lapply(out_null_CLM$diagnostic.res3, function(x) x$post$postmean))
out_null_CLM_pseudo_z = unlist(lapply(out_null_CLM$diagnostic.res3, function(x) x$post$post_z))</code></pre>
<p>MLE <span class="math inline">\(s^2\)</span>:</p>
<pre class="r"><code>par(mfrow=c(2,3))
# MLE
diff_sample = z-out_null_sample_mean; diff_IBS = z-out_null_IBS_mean; diff_CLM = z-out_null_CLM_mean; 
ymax = ceiling(max(c(diff_sample, diff_IBS, diff_CLM))); ymin = floor(min(c(diff_sample, diff_IBS, diff_CLM)))
plot(diff_sample, ylab=&#39;difference using GBR LD&#39;, main=&#39;Difference (z from GBR)&#39;,cex=0.6, ylim = c(ymin, ymax))
plot(diff_IBS, ylab=&#39;difference using IBS LD&#39;, ylim = c(ymin, ymax))
plot(diff_CLM, ylab=&#39;difference using CLM LD&#39;, ylim = c(ymin, ymax))

ymax = ceiling(max(c(out_null_sample_z, out_null_IBS_z, out_null_CLM_z))); ymin = floor(min(c(out_null_sample_z, out_null_IBS_z, out_null_CLM_z)))
plot(out_null_sample_z, ylab=&#39;standardized difference using GBR LD&#39;, main=&#39;Standardized Difference&#39;,cex=0.6, ylim = c(ymin, ymax))
plot(out_null_IBS_z, ylab=&#39;standardized difference using IBS LD&#39;, ylim = c(ymin, ymax))
plot(out_null_CLM_z, ylab=&#39;standardized difference using CLM LD&#39;, ylim = c(ymin, ymax))</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-7-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Truncated <span class="math inline">\(s^2\)</span>:</p>
<pre class="r"><code>par(mfrow=c(2,3))
# truncate
diff_sample = z-out_null_sample_trun_mean; diff_IBS = z-out_null_IBS_trun_mean; diff_CLM = z-out_null_CLM_trun_mean; 
ymax = ceiling(max(c(diff_sample, diff_IBS, diff_CLM))); ymin = floor(min(c(diff_sample, diff_IBS, diff_CLM)))
plot(diff_sample, ylab=&#39;difference using GBR LD&#39;, main=&#39;Difference (z from GBR)&#39;,cex=0.6, ylim = c(ymin, ymax))
plot(diff_IBS, ylab=&#39;difference using IBS LD&#39;, ylim = c(ymin, ymax))
plot(diff_CLM, ylab=&#39;difference using CLM LD&#39;, ylim = c(ymin, ymax))

ymax = ceiling(max(c(out_null_sample_trun_z, out_null_IBS_trun_z, out_null_CLM_trun_z))); ymin = floor(min(c(out_null_sample_trun_z, out_null_IBS_trun_z, out_null_CLM_trun_z)))
plot(out_null_sample_trun_z, ylab=&#39;standardized difference using GBR LD&#39;, main=&#39;Standardized Difference&#39;,cex=0.6, ylim = c(ymin, ymax))
plot(out_null_IBS_trun_z, ylab=&#39;standardized difference using IBS LD&#39;, ylim = c(ymin, ymax))
plot(out_null_CLM_trun_z, ylab=&#39;standardized difference using CLM LD&#39;, ylim = c(ymin, ymax))</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-8-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Pseudo likelihood <span class="math inline">\(s^2\)</span>:</p>
<pre class="r"><code>par(mfrow=c(2,3))
# pseudo
diff_sample = z-out_null_sample_pseudo_mean; diff_IBS = z-out_null_IBS_pseudo_mean; diff_CLM = z-out_null_CLM_pseudo_mean; 
ymax = ceiling(max(c(diff_sample, diff_IBS, diff_CLM))); ymin = floor(min(c(diff_sample, diff_IBS, diff_CLM)))
plot(diff_sample, ylab=&#39;difference using GBR LD&#39;, main=&#39;Difference (z from GBR)&#39;,cex=0.6, ylim = c(ymin, ymax))
plot(diff_IBS, ylab=&#39;difference using IBS LD&#39;, ylim = c(ymin, ymax))
plot(diff_CLM, ylab=&#39;difference using CLM LD&#39;, ylim = c(ymin, ymax))

ymax = ceiling(max(c(out_null_sample_pseudo_z, out_null_IBS_pseudo_z, out_null_CLM_pseudo_z))); ymin = floor(min(c(out_null_sample_pseudo_z, out_null_IBS_pseudo_z, out_null_CLM_pseudo_z)))
plot(out_null_sample_pseudo_z, ylab=&#39;standardized difference using GBR LD&#39;, main=&#39;Standardized Difference&#39;,cex=0.6, ylim = c(ymin, ymax))
plot(out_null_IBS_pseudo_z, ylab=&#39;standardized difference using IBS LD&#39;, ylim = c(ymin, ymax))
plot(out_null_CLM_pseudo_z, ylab=&#39;standardized difference using CLM LD&#39;, ylim = c(ymin, ymax))</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-9-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="simulation-under-null-z-0-with-one-flipped-sign" class="section level2">
<h2>Simulation under null (z = 0) with one flipped sign</h2>
<pre class="r"><code>out_null_flip_sample = out_null_flip %&gt;% filter(susie_rss.ld_type == &#39;sample&#39;, susie_rss.correct_ld == FALSE)
out_null_flip_IBS = out_null_flip %&gt;% filter(susie_rss.ld_type == &#39;IBS&#39;, susie_rss.correct_ld == FALSE)
out_null_flip_CLM = out_null_flip %&gt;% filter(susie_rss.ld_type == &#39;CLM&#39;, susie_rss.correct_ld == FALSE)</code></pre>
<div id="check-s2-1" class="section level3">
<h3>Check <span class="math inline">\(s^2\)</span></h3>
<pre class="r"><code>par(mfrow=c(2,3))
out_null_flip_sample_s2 = apply(out_null_flip_sample, 1, function(x) x$diagnostic.res1_simple$s2)
out_null_flip_sample_s2_pseudo = apply(out_null_flip_sample, 1, function(x) x$diagnostic.res3$s2)
out_null_flip_IBS_s2 = apply(out_null_flip_IBS, 1, function(x) x$diagnostic.res1_simple$s2)
out_null_flip_IBS_s2_pseudo = apply(out_null_flip_IBS, 1, function(x) x$diagnostic.res3$s2)
out_null_flip_CLM_s2 = apply(out_null_flip_CLM, 1, function(x) x$diagnostic.res1_simple$s2)
out_null_flip_CLM_s2_pseudo = apply(out_null_flip_CLM, 1, function(x) x$diagnostic.res3$s2)

plot(out_null_flip_sample_s2, out_null_flip_IBS_s2, xlab = &#39;LD = GBR&#39;, ylab=&#39;LD = IBS&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)
plot(out_null_flip_sample_s2, out_null_flip_CLM_s2, xlab = &#39;LD = GBR&#39;, ylab=&#39;LD = CLM&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)
plot(out_null_flip_IBS_s2, out_null_flip_CLM_s2, xlab = &#39;LD = IBS&#39;, ylab=&#39;LD = CLM&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)

plot(out_null_flip_sample_s2, out_null_flip_sample_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;pseudo likelihood&#39;, main=&#39;GBR&#39;)
abline(0,1)
plot(out_null_flip_IBS_s2, out_null_flip_IBS_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;pseudo likelihood&#39;, main=&#39;IBS&#39;)
abline(0,1)
plot(out_null_flip_CLM_s2, out_null_flip_CLM_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;pseudo likelihood&#39;, main=&#39;CLM&#39;)
abline(0,1)</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-11-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Comparing the estimated MLE <span class="math inline">\(s^2\)</span> with non-flipped sign:</p>
<pre class="r"><code>plot(out_null_sample_s2, out_null_flip_sample_s2, xlab=&#39;non-flip&#39;, ylab=&#39;flip&#39;, main=&#39;GBR s2&#39;)
abline(0,1)</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-12-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="check-conditional-mean-1" class="section level3">
<h3>Check conditional mean</h3>
<pre class="r"><code>z_flip = unlist(lapply(out_null_flip_sample$diagnostic.res1_simple, function(x) x$post$z))
out_null_flip_sample_mean = unlist(lapply(out_null_flip_sample$diagnostic.res1_simple, function(x) x$post$postmean))
out_null_flip_sample_z = unlist(lapply(out_null_flip_sample$diagnostic.res1_simple, function(x) x$post$post_z))
out_null_flip_IBS_mean = unlist(lapply(out_null_flip_IBS$diagnostic.res1_simple, function(x) x$post$postmean))
out_null_flip_IBS_z = unlist(lapply(out_null_flip_IBS$diagnostic.res1_simple, function(x) x$post$post_z))
out_null_flip_CLM_mean = unlist(lapply(out_null_flip_CLM$diagnostic.res1_simple, function(x) x$post$postmean))
out_null_flip_CLM_z = unlist(lapply(out_null_flip_CLM$diagnostic.res1_simple, function(x) x$post$post_z))

out_null_flip_sample_trun_mean = unlist(lapply(out_null_flip_sample$diagnostic.res2, function(x) x$post$postmean))
out_null_flip_sample_trun_z = unlist(lapply(out_null_flip_sample$diagnostic.res2, function(x) x$post$post_z))
out_null_flip_IBS_trun_mean = unlist(lapply(out_null_flip_IBS$diagnostic.res2, function(x) x$post$postmean))
out_null_flip_IBS_trun_z = unlist(lapply(out_null_flip_IBS$diagnostic.res2, function(x) x$post$post_z))
out_null_flip_CLM_trun_mean = unlist(lapply(out_null_flip_CLM$diagnostic.res2, function(x) x$post$postmean))
out_null_flip_CLM_trun_z = unlist(lapply(out_null_flip_CLM$diagnostic.res2, function(x) x$post$post_z))

out_null_flip_sample_pseudo_mean = unlist(lapply(out_null_flip_sample$diagnostic.res3, function(x) x$post$postmean))
out_null_flip_sample_pseudo_z = unlist(lapply(out_null_flip_sample$diagnostic.res3, function(x) x$post$post_z))
out_null_flip_IBS_pseudo_mean = unlist(lapply(out_null_flip_IBS$diagnostic.res3, function(x) x$post$postmean))
out_null_flip_IBS_pseudo_z = unlist(lapply(out_null_flip_IBS$diagnostic.res3, function(x) x$post$post_z))
out_null_flip_CLM_pseudo_mean = unlist(lapply(out_null_flip_CLM$diagnostic.res3, function(x) x$post$postmean))
out_null_flip_CLM_pseudo_z = unlist(lapply(out_null_flip_CLM$diagnostic.res3, function(x) x$post$post_z))</code></pre>
<p>MLE <span class="math inline">\(s^2\)</span>:</p>
<pre class="r"><code>par(mfrow=c(2,3))
idx = which(z != z_flip)
diff_sample = z_flip-out_null_flip_sample_mean; diff_IBS = z_flip-out_null_flip_IBS_mean; diff_CLM = z_flip-out_null_flip_CLM_mean;
ymax = ceiling(max(c(diff_sample, diff_IBS, diff_CLM))); ymin = floor(min(c(diff_sample, diff_IBS, diff_CLM)))
plot(diff_sample, ylab=&#39;difference using GBR LD&#39;, main=&#39;Difference (z from GBR)&#39;, cex=0.6, ylim = c(ymin, ymax))
points(idx, diff_sample[idx], col=&#39;red&#39;, pch=16)
plot(diff_IBS, ylab=&#39;difference using IBS LD&#39;, ylim = c(ymin, ymax))
points(idx, diff_IBS[idx], col=&#39;red&#39;, pch=16)
plot(diff_CLM, ylab=&#39;difference using CLM LD&#39;, ylim = c(ymin, ymax))
points(idx, diff_CLM[idx], col=&#39;red&#39;, pch=16)

ymax = ceiling(max(c(out_null_flip_sample_z, out_null_flip_IBS_z, out_null_flip_CLM_z))); ymin = floor(min(c(out_null_flip_sample_z, out_null_flip_IBS_z, out_null_flip_CLM_z)))
plot(out_null_flip_sample_z, ylab=&#39;standardized difference using GBR LD&#39;, main=&#39;Standardized Difference&#39;, cex=0.6, ylim = c(ymin, ymax))
points(idx, out_null_flip_sample_z[idx], col=&#39;red&#39;, pch=16)
plot(out_null_flip_IBS_z, ylab=&#39;standardized difference using IBS LD&#39;, ylim = c(ymin, ymax))
points(idx, out_null_flip_IBS_z[idx], col=&#39;red&#39;, pch=16)
plot(out_null_flip_CLM_z, ylab=&#39;standardized difference using CLM LD&#39;, ylim = c(ymin, ymax))
points(idx, out_null_flip_CLM_z[idx], col=&#39;red&#39;, pch=16)</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-14-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Truncated <span class="math inline">\(s^2\)</span>:</p>
<pre class="r"><code>par(mfrow=c(2,3))
diff_sample = z_flip-out_null_flip_sample_trun_mean; diff_IBS = z_flip-out_null_flip_IBS_trun_mean; diff_CLM = z_flip-out_null_flip_CLM_trun_mean;
ymax = ceiling(max(c(diff_sample, diff_IBS, diff_CLM))); ymin = floor(min(c(diff_sample, diff_IBS, diff_CLM)))
plot(diff_sample, ylab=&#39;difference using GBR LD&#39;, main=&#39;Difference (z from GBR)&#39;, cex=0.6, ylim = c(ymin, ymax))
points(idx, diff_sample[idx], col=&#39;red&#39;, pch=16)
plot(diff_IBS, ylab=&#39;difference using IBS LD&#39;, ylim = c(ymin, ymax))
points(idx, diff_IBS[idx], col=&#39;red&#39;, pch=16)
plot(diff_CLM, ylab=&#39;difference using CLM LD&#39;, ylim = c(ymin, ymax))
points(idx, diff_CLM[idx], col=&#39;red&#39;, pch=16)

ymax = ceiling(max(c(out_null_flip_sample_trun_z, out_null_flip_IBS_trun_z, out_null_flip_CLM_trun_z))); ymin = floor(min(c(out_null_flip_sample_trun_z, out_null_flip_IBS_trun_z, out_null_flip_CLM_trun_z)))
plot(out_null_flip_sample_trun_z, ylab=&#39;standardized difference using GBR LD&#39;, main=&#39;Standardized Difference&#39;, cex=0.6, ylim = c(ymin, ymax))
points(idx, out_null_flip_sample_trun_z[idx], col=&#39;red&#39;, pch=16)
plot(out_null_flip_IBS_trun_z, ylab=&#39;standardized difference using IBS LD&#39;, ylim = c(ymin, ymax))
points(idx, out_null_flip_IBS_trun_z[idx], col=&#39;red&#39;, pch=16)
plot(out_null_flip_CLM_trun_z, ylab=&#39;standardized difference using CLM LD&#39;, ylim = c(ymin, ymax))
points(idx, out_null_flip_CLM_trun_z[idx], col=&#39;red&#39;, pch=16)</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-15-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Pseudo likelihood <span class="math inline">\(s^2\)</span>:</p>
<pre class="r"><code>par(mfrow=c(2,3))
idx = which(z != z_flip)
diff_sample = z_flip-out_null_flip_sample_pseudo_mean; diff_IBS = z_flip-out_null_flip_IBS_pseudo_mean; diff_CLM = z_flip-out_null_flip_CLM_pseudo_mean;
ymax = ceiling(max(c(diff_sample, diff_IBS, diff_CLM))); ymin = floor(min(c(diff_sample, diff_IBS, diff_CLM)))
plot(diff_sample, ylab=&#39;difference using GBR LD&#39;, main=&#39;Difference (z from GBR)&#39;, cex=0.6, ylim = c(ymin, ymax))
points(idx, diff_sample[idx], col=&#39;red&#39;, pch=16)
plot(diff_IBS, ylab=&#39;difference using IBS LD&#39;, ylim = c(ymin, ymax))
points(idx, diff_IBS[idx], col=&#39;red&#39;, pch=16)
plot(diff_CLM, ylab=&#39;difference using CLM LD&#39;, ylim = c(ymin, ymax))
points(idx, diff_CLM[idx], col=&#39;red&#39;, pch=16)

ymax = ceiling(max(c(out_null_flip_sample_pseudo_z, out_null_flip_IBS_pseudo_z, out_null_flip_CLM_pseudo_z))); ymin = floor(min(c(out_null_flip_sample_pseudo_z, out_null_flip_IBS_pseudo_z, out_null_flip_CLM_pseudo_z)))
plot(out_null_flip_sample_pseudo_z, ylab=&#39;standardized difference using GBR LD&#39;, main=&#39;Standardized Difference&#39;, cex=0.6, ylim = c(ymin, ymax))
points(idx, out_null_flip_sample_pseudo_z[idx], col=&#39;red&#39;, pch=16)
plot(out_null_flip_IBS_z, ylab=&#39;standardized difference using IBS LD&#39;, ylim = c(ymin, ymax))
points(idx, out_null_flip_IBS_pseudo_z[idx], col=&#39;red&#39;, pch=16)
plot(out_null_flip_CLM_z, ylab=&#39;standardized difference using CLM LD&#39;, ylim = c(ymin, ymax))
points(idx, out_null_flip_CLM_pseudo_z[idx], col=&#39;red&#39;, pch=16)</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-16-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="simulation-with-signal" class="section level2">
<h2>Simulation with signal</h2>
<pre class="r"><code>out_sample = out_signal %&gt;% filter(susie_rss.ld_type == &#39;sample&#39;, susie_rss.correct_ld == FALSE)
out_IBS = out_signal %&gt;% filter(susie_rss.ld_type == &#39;IBS&#39;, susie_rss.correct_ld == FALSE)
out_CLM = out_signal %&gt;% filter(susie_rss.ld_type == &#39;CLM&#39;, susie_rss.correct_ld == FALSE)
out_IBSz = out_signal %&gt;% filter(susie_rss.ld_type == &#39;IBS&#39;, susie_rss.correct_ld == TRUE)
out_CLMz = out_signal %&gt;% filter(susie_rss.ld_type == &#39;CLM&#39;, susie_rss.correct_ld == TRUE)</code></pre>
<div id="check-s2-2" class="section level3">
<h3>Check <span class="math inline">\(s^2\)</span></h3>
<pre class="r"><code>par(mfrow=c(2,3))
out_sample_s2 = apply(out_sample, 1, function(x) x$diagnostic.res1_simple$s2)
out_sample_s2_pseudo = apply(out_sample, 1, function(x) x$diagnostic.res3$s2)
out_IBS_s2 = apply(out_IBS, 1, function(x) x$diagnostic.res1_simple$s2)
out_IBS_s2_pseudo = apply(out_IBS, 1, function(x) x$diagnostic.res3$s2)
out_CLM_s2 = apply(out_CLM, 1, function(x) x$diagnostic.res1_simple$s2)
out_CLM_s2_pseudo = apply(out_CLM, 1, function(x) x$diagnostic.res3$s2)
# out_IBSz_s2 = apply(out_IBSz, 1, function(x) x$diagnostic.res1_simple$s2)
# out_CLMz_s2 = apply(out_CLMz, 1, function(x) x$diagnostic.res1_simple$s2)

plot(out_sample_s2, out_IBS_s2, xlab = &#39;LD = GBR&#39;, ylab=&#39;LD = IBS&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)
plot(out_sample_s2, out_CLM_s2, xlab = &#39;LD = GBR&#39;, ylab=&#39;LD = CLM&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)
plot(out_IBS_s2, out_CLM_s2, xlab = &#39;LD = IBS&#39;, ylab=&#39;LD = CLM&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)

plot(out_sample_s2, out_sample_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;psudo likelihood&#39;, main=&#39;GBR&#39;)
abline(0,1)
plot(out_IBS_s2, out_IBS_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;psudo likelihood&#39;, main=&#39;IBS&#39;)
abline(0,1)
plot(out_CLM_s2, out_CLM_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;psudo likelihood&#39;, main=&#39;CLM&#39;)
abline(0,1)</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-18-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="check-conditional-mean-2" class="section level3">
<h3>Check conditional mean</h3>
<pre class="r"><code>z = unlist(lapply(out_sample$diagnostic.res1_simple, function(x) x$post$z))
out_sample_mean = unlist(lapply(out_sample$diagnostic.res1_simple, function(x) x$post$postmean))
out_sample_z = unlist(lapply(out_sample$diagnostic.res1_simple, function(x) x$post$post_z))
# out_IBS_mean = unlist(lapply(out_IBS$diagnostic.res1_simple, function(x) x$post$postmean))
# out_IBS_z = unlist(lapply(out_IBS$diagnostic.res1_simple, function(x) x$post$post_z))
# out_CLM_mean = unlist(lapply(out_CLM$diagnostic.res1_simple, function(x) x$post$postmean))
# out_CLM_z = unlist(lapply(out_CLM$diagnostic.res1_simple, function(x) x$post$post_z))

out_sample_trun_mean = unlist(lapply(out_sample$diagnostic.res2, function(x) x$post$postmean))
out_sample_trun_z = unlist(lapply(out_sample$diagnostic.res2, function(x) x$post$post_z))
# out_IBS_trun_mean = unlist(lapply(out_IBS$diagnostic.res2, function(x) x$post$postmean))
# out_IBS_trun_z = unlist(lapply(out_IBS$diagnostic.res2, function(x) x$post$post_z))
# out_CLM_trun_mean = unlist(lapply(out_CLM$diagnostic.res2, function(x) x$post$postmean))
# out_CLM_trun_z = unlist(lapply(out_CLM$diagnostic.res2, function(x) x$post$post_z))

out_sample_pseudo_mean = unlist(lapply(out_sample$diagnostic.res3, function(x) x$post$postmean))
out_sample_pseudo_z = unlist(lapply(out_sample$diagnostic.res3, function(x) x$post$post_z))
# out_IBS_pseudo_mean = unlist(lapply(out_IBS$diagnostic.res3, function(x) x$post$postmean))
# out_IBS_pseudo_z = unlist(lapply(out_IBS$diagnostic.res3, function(x) x$post$post_z))
# out_CLM_pseudo_mean = unlist(lapply(out_CLM$diagnostic.res3, function(x) x$post$postmean))
# out_CLM_pseudo_z = unlist(lapply(out_CLM$diagnostic.res3, function(x) x$post$post_z))</code></pre>
<pre class="r"><code>par(mfrow=c(2,3))
diff_mle = z-out_sample_mean; diff_trun = z-out_sample_trun_mean; diff_pseudo = z-out_sample_pseudo_mean; 
ymax = ceiling(max(c(diff_mle, diff_trun, diff_pseudo))); ymin = floor(min(c(diff_mle, diff_trun, diff_pseudo)))
plot(diff_mle, ylab=&#39;difference&#39;, main=&#39;LD=GBR; MLE s2&#39;,cex=0.6, ylim = c(ymin, ymax))
plot(diff_trun, ylab=&#39;difference&#39;, main=&#39;Truncated s2&#39;, ylim = c(ymin, ymax))
plot(diff_pseudo, ylab=&#39;difference&#39;, main=&#39;Pseudo L s2&#39;, ylim = c(ymin, ymax))

ymax = ceiling(max(c(out_sample_z, out_sample_trun_z, out_sample_pseudo_z))); ymin = floor(min(c(out_sample_z, out_sample_trun_z, out_sample_pseudo_z)))
plot(out_sample_z, ylab=&#39;standardized difference&#39;, main=&#39;MLE s2&#39;,cex=0.6, ylim = c(ymin, ymax))
plot(out_sample_trun_z, ylab=&#39;standardized difference&#39;, main=&#39;Truncated s2&#39;,ylim = c(ymin, ymax))
plot(out_sample_pseudo_z, ylab=&#39;standardized difference&#39;, main=&#39;Pseudo L s2&#39;,ylim = c(ymin, ymax))</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-20-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="simulation-with-signal-and-flipped-sign-at-signal" class="section level2">
<h2>Simulation with signal and flipped sign at signal</h2>
<pre class="r"><code>out_flip_sample = out_signal_flip_signal %&gt;% filter(susie_rss.ld_type == &#39;sample&#39;, susie_rss.correct_ld == FALSE)
out_flip_IBS = out_signal_flip_signal %&gt;% filter(susie_rss.ld_type == &#39;IBS&#39;, susie_rss.correct_ld == FALSE)
out_flip_CLM = out_signal_flip_signal %&gt;% filter(susie_rss.ld_type == &#39;CLM&#39;, susie_rss.correct_ld == FALSE)</code></pre>
<div id="check-s2-3" class="section level3">
<h3>Check <span class="math inline">\(s^2\)</span></h3>
<pre class="r"><code>par(mfrow=c(2,3))
out_flip_sample_s2 = apply(out_flip_sample, 1, function(x) x$diagnostic.res1_simple$s2)
out_flip_sample_s2_pseudo = apply(out_flip_sample, 1, function(x) x$diagnostic.res3$s2)
out_flip_IBS_s2 = apply(out_flip_IBS, 1, function(x) x$diagnostic.res1_simple$s2)
out_flip_IBS_s2_pseudo = apply(out_flip_IBS, 1, function(x) x$diagnostic.res3$s2)
out_flip_CLM_s2 = apply(out_flip_CLM, 1, function(x) x$diagnostic.res1_simple$s2)
out_flip_CLM_s2_pseudo = apply(out_flip_CLM, 1, function(x) x$diagnostic.res3$s2)

plot(out_flip_sample_s2, out_flip_IBS_s2, xlab = &#39;LD = GBR&#39;, ylab=&#39;LD = IBS&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)
plot(out_flip_sample_s2, out_flip_CLM_s2, xlab = &#39;LD = GBR&#39;, ylab=&#39;LD = CLM&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)
plot(out_flip_IBS_s2, out_flip_CLM_s2, xlab = &#39;LD = IBS&#39;, ylab=&#39;LD = CLM&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)

plot(out_flip_sample_s2, out_flip_sample_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;psudo likelihood&#39;, main=&#39;GBR&#39;)
abline(0,1)
plot(out_flip_IBS_s2, out_flip_IBS_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;psudo likelihood&#39;, main=&#39;IBS&#39;)
abline(0,1)
plot(out_flip_CLM_s2, out_flip_CLM_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;psudo likelihood&#39;, main=&#39;CLM&#39;)
abline(0,1)</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-22-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="check-conditional-mean-3" class="section level3">
<h3>Check conditional mean</h3>
<pre class="r"><code>z_flip = unlist(lapply(out_flip_sample$diagnostic.res1_simple, function(x) x$post$z))
out_flip_sample_mean = unlist(lapply(out_flip_sample$diagnostic.res1_simple, function(x) x$post$postmean))
out_flip_sample_z = unlist(lapply(out_flip_sample$diagnostic.res1_simple, function(x) x$post$post_z))
# out_flip_IBS_mean = unlist(lapply(out_flip_IBS$diagnostic.res1_simple, function(x) x$post$postmean))
# out_flip_IBS_z = unlist(lapply(out_flip_IBS$diagnostic.res1_simple, function(x) x$post$post_z))
# out_flip_CLM_mean = unlist(lapply(out_flip_CLM$diagnostic.res1_simple, function(x) x$post$postmean))
# out_flip_CLM_z = unlist(lapply(out_flip_CLM$diagnostic.res1_simple, function(x) x$post$post_z))

out_flip_sample_trun_mean = unlist(lapply(out_flip_sample$diagnostic.res2, function(x) x$post$postmean))
out_flip_sample_trun_z = unlist(lapply(out_flip_sample$diagnostic.res2, function(x) x$post$post_z))
# out_flip_IBS_trun_mean = unlist(lapply(out_flip_IBS$diagnostic.res2, function(x) x$post$postmean))
# out_flip_IBS_trun_z = unlist(lapply(out_flip_IBS$diagnostic.res2, function(x) x$post$post_z))
# out_flip_CLM_trun_mean = unlist(lapply(out_flip_CLM$diagnostic.res2, function(x) x$post$postmean))
# out_flip_CLM_trun_z = unlist(lapply(out_flip_CLM$diagnostic.res2, function(x) x$post$post_z))

out_flip_sample_pseudo_mean = unlist(lapply(out_flip_sample$diagnostic.res3, function(x) x$post$postmean))
out_flip_sample_pseudo_z = unlist(lapply(out_flip_sample$diagnostic.res3, function(x) x$post$post_z))
# out_flip_IBS_pseudo_mean = unlist(lapply(out_flip_IBS$diagnostic.res3, function(x) x$post$postmean))
# out_flip_IBS_pseudo_z = unlist(lapply(out_flip_IBS$diagnostic.res3, function(x) x$post$post_z))
# out_flip_CLM_pseudo_mean = unlist(lapply(out_flip_CLM$diagnostic.res3, function(x) x$post$postmean))
# out_flip_CLM_pseudo_z = unlist(lapply(out_flip_CLM$diagnostic.res3, function(x) x$post$post_z))</code></pre>
<pre class="r"><code>par(mfrow=c(2,3))
idx = which(z != z_flip)
diff_mle = z_flip-out_flip_sample_mean; diff_trun = z_flip-out_flip_sample_trun_mean; diff_pseudo = z_flip-out_flip_sample_pseudo_mean; 
ymax = ceiling(max(c(diff_mle, diff_trun, diff_pseudo))); ymin = floor(min(c(diff_mle, diff_trun, diff_pseudo)))
plot(diff_mle, ylab=&#39;difference&#39;, main=&#39;LD=GBR; MLE s2&#39;,cex=0.6, ylim = c(ymin, ymax))
points(idx, diff_mle[idx], col=&#39;red&#39;, pch=16)
plot(diff_trun, ylab=&#39;difference&#39;, main=&#39;Truncated s2&#39;, ylim = c(ymin, ymax))
points(idx, diff_trun[idx], col=&#39;red&#39;, pch=16)
plot(diff_pseudo, ylab=&#39;difference&#39;, main=&#39;Pseudo L s2&#39;, ylim = c(ymin, ymax))
points(idx, diff_pseudo[idx], col=&#39;red&#39;, pch=16)

ymax = ceiling(max(c(out_flip_sample_z, out_flip_sample_trun_z, out_flip_sample_pseudo_z))); ymin = floor(min(c(out_flip_sample_z, out_flip_sample_trun_z, out_flip_sample_pseudo_z)))
plot(out_flip_sample_z, ylab=&#39;standardized difference&#39;, main=&#39;MLE s2&#39;,cex=0.6, ylim = c(ymin, ymax))
points(idx, out_flip_sample_z[idx], col=&#39;red&#39;, pch=16)
plot(out_flip_sample_trun_z, ylab=&#39;standardized difference&#39;, main=&#39;Truncated s2&#39;,ylim = c(ymin, ymax))
points(idx, out_flip_sample_trun_z[idx], col=&#39;red&#39;, pch=16)
plot(out_flip_sample_pseudo_z, ylab=&#39;standardized difference&#39;, main=&#39;Pseudo L s2&#39;,ylim = c(ymin, ymax))
points(idx, out_flip_sample_pseudo_z[idx], col=&#39;red&#39;, pch=16)</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-24-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="simulation-with-signal-and-flipped-sign-at-null" class="section level2">
<h2>Simulation with signal and flipped sign at null</h2>
<pre class="r"><code>out_flip_null_sample = out_signal_flip_null %&gt;% filter(susie_rss.ld_type == &#39;sample&#39;, susie_rss.correct_ld == FALSE)
out_flip_null_IBS = out_signal_flip_null %&gt;% filter(susie_rss.ld_type == &#39;IBS&#39;, susie_rss.correct_ld == FALSE)
out_flip_null_CLM = out_signal_flip_null %&gt;% filter(susie_rss.ld_type == &#39;CLM&#39;, susie_rss.correct_ld == FALSE)</code></pre>
<div id="check-s2-4" class="section level3">
<h3>Check <span class="math inline">\(s^2\)</span></h3>
<pre class="r"><code>par(mfrow=c(2,3))
out_flip_null_sample_s2 = apply(out_flip_null_sample, 1, function(x) x$diagnostic.res1_simple$s2)
out_flip_null_sample_s2_pseudo = apply(out_flip_null_sample, 1, function(x) x$diagnostic.res3$s2)
out_flip_null_IBS_s2 = apply(out_flip_null_IBS, 1, function(x) x$diagnostic.res1_simple$s2)
out_flip_null_IBS_s2_pseudo = apply(out_flip_null_IBS, 1, function(x) x$diagnostic.res3$s2)
out_flip_null_CLM_s2 = apply(out_flip_null_CLM, 1, function(x) x$diagnostic.res1_simple$s2)
out_flip_null_CLM_s2_pseudo = apply(out_flip_null_CLM, 1, function(x) x$diagnostic.res3$s2)

plot(out_flip_null_sample_s2, out_flip_null_IBS_s2, xlab = &#39;LD = GBR&#39;, ylab=&#39;LD = IBS&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)
plot(out_flip_null_sample_s2, out_flip_null_CLM_s2, xlab = &#39;LD = GBR&#39;, ylab=&#39;LD = CLM&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)
plot(out_flip_null_IBS_s2, out_flip_null_CLM_s2, xlab = &#39;LD = IBS&#39;, ylab=&#39;LD = CLM&#39;, main=&#39;GBR z scores&#39;)
abline(0,1)

plot(out_flip_null_sample_s2, out_flip_null_sample_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;psudo likelihood&#39;, main=&#39;GBR&#39;)
abline(0,1)
plot(out_flip_null_IBS_s2, out_flip_null_IBS_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;psudo likelihood&#39;, main=&#39;IBS&#39;)
abline(0,1)
plot(out_flip_null_CLM_s2, out_flip_null_CLM_s2_pseudo, xlab = &#39;MLE&#39;, ylab=&#39;psudo likelihood&#39;, main=&#39;CLM&#39;)
abline(0,1)</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-26-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="check-conditional-mean-4" class="section level3">
<h3>Check conditional mean</h3>
<pre class="r"><code>z_flip = unlist(lapply(out_flip_null_sample$diagnostic.res1_simple, function(x) x$post$z))
out_flip_null_sample_mean = unlist(lapply(out_flip_null_sample$diagnostic.res1_simple, function(x) x$post$postmean))
out_flip_null_sample_z = unlist(lapply(out_flip_null_sample$diagnostic.res1_simple, function(x) x$post$post_z))
# out_flip_null_IBS_mean = unlist(lapply(out_flip_null_IBS$diagnostic.res1_simple, function(x) x$post$postmean))
# out_flip_null_IBS_z = unlist(lapply(out_flip_null_IBS$diagnostic.res1_simple, function(x) x$post$post_z))
# out_flip_null_CLM_mean = unlist(lapply(out_flip_null_CLM$diagnostic.res1_simple, function(x) x$post$postmean))
# out_flip_null_CLM_z = unlist(lapply(out_flip_null_CLM$diagnostic.res1_simple, function(x) x$post$post_z))

out_flip_null_sample_trun_mean = unlist(lapply(out_flip_null_sample$diagnostic.res2, function(x) x$post$postmean))
out_flip_null_sample_trun_z = unlist(lapply(out_flip_null_sample$diagnostic.res2, function(x) x$post$post_z))
# out_flip_null_IBS_trun_mean = unlist(lapply(out_flip_null_IBS$diagnostic.res2, function(x) x$post$postmean))
# out_flip_null_IBS_trun_z = unlist(lapply(out_flip_null_IBS$diagnostic.res2, function(x) x$post$post_z))
# out_flip_null_CLM_trun_mean = unlist(lapply(out_flip_null_CLM$diagnostic.res2, function(x) x$post$postmean))
# out_flip_null_CLM_trun_z = unlist(lapply(out_flip_null_CLM$diagnostic.res2, function(x) x$post$post_z))

out_flip_null_sample_pseudo_mean = unlist(lapply(out_flip_null_sample$diagnostic.res3, function(x) x$post$postmean))
out_flip_null_sample_pseudo_z = unlist(lapply(out_flip_null_sample$diagnostic.res3, function(x) x$post$post_z))
# out_flip_null_IBS_pseudo_mean = unlist(lapply(out_flip_null_IBS$diagnostic.res3, function(x) x$post$postmean))
# out_flip_null_IBS_pseudo_z = unlist(lapply(out_flip_null_IBS$diagnostic.res3, function(x) x$post$post_z))
# out_flip_null_CLM_pseudo_mean = unlist(lapply(out_flip_null_CLM$diagnostic.res3, function(x) x$post$postmean))
# out_flip_null_CLM_pseudo_z = unlist(lapply(out_flip_null_CLM$diagnostic.res3, function(x) x$post$post_z))</code></pre>
<pre class="r"><code>par(mfrow=c(2,3))
idx = which(z != z_flip)
diff_mle = z_flip-out_flip_null_sample_mean; diff_trun = z_flip-out_flip_null_sample_trun_mean; diff_pseudo = z_flip-out_flip_null_sample_pseudo_mean; 
ymax = ceiling(max(c(diff_mle, diff_trun, diff_pseudo))); ymin = floor(min(c(diff_mle, diff_trun, diff_pseudo)))
plot(diff_mle, ylab=&#39;difference&#39;, main=&#39;LD=GBR; MLE s2&#39;,cex=0.6, ylim = c(ymin, ymax))
points(idx, diff_mle[idx], col=&#39;red&#39;, pch=16)
plot(diff_trun, ylab=&#39;difference&#39;, main=&#39;Truncated s2&#39;, ylim = c(ymin, ymax))
points(idx, diff_trun[idx], col=&#39;red&#39;, pch=16)
plot(diff_pseudo, ylab=&#39;difference&#39;, main=&#39;Pseudo L s2&#39;, ylim = c(ymin, ymax))
points(idx, diff_pseudo[idx], col=&#39;red&#39;, pch=16)

ymax = ceiling(max(c(out_flip_null_sample_z, out_flip_null_sample_trun_z, out_flip_null_sample_pseudo_z))); ymin = floor(min(c(out_flip_null_sample_z, out_flip_null_sample_trun_z, out_flip_null_sample_pseudo_z)))
plot(out_flip_null_sample_z, ylab=&#39;standardized difference&#39;, main=&#39;MLE s2&#39;,cex=0.6, ylim = c(ymin, ymax))
points(idx, out_flip_null_sample_z[idx], col=&#39;red&#39;, pch=16)
plot(out_flip_null_sample_trun_z, ylab=&#39;standardized difference&#39;, main=&#39;Truncated s2&#39;,ylim = c(ymin, ymax))
points(idx, out_flip_null_sample_trun_z[idx], col=&#39;red&#39;, pch=16)
plot(out_flip_null_sample_pseudo_z, ylab=&#39;standardized difference&#39;, main=&#39;Pseudo L s2&#39;,ylim = c(ymin, ymax))
points(idx, out_flip_null_sample_pseudo_z[idx], col=&#39;red&#39;, pch=16)</code></pre>
<p><img src="figure/diagnostic.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/3e5c00198caf825e243980b70c9eea5854d0ab47/docs/figure/diagnostic.Rmd/unnamed-chunk-28-1.png" target="_blank">3e5c001</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-01-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.0.2     tibble_3.0.4    workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5       rstudioapi_0.13  whisker_0.4      knitr_1.30      
 [5] magrittr_2.0.1   tidyselect_1.1.0 R6_2.5.0         rlang_0.4.10    
 [9] stringr_1.4.0    tools_4.0.3      xfun_0.19        git2r_0.27.1    
[13] htmltools_0.5.0  ellipsis_0.3.1   rprojroot_2.0.2  yaml_2.2.1      
[17] digest_0.6.27    lifecycle_0.2.0  crayon_1.3.4     purrr_0.3.4     
[21] later_1.1.0.1    vctrs_0.3.6      promises_1.1.1   fs_1.5.0        
[25] glue_1.4.2       evaluate_0.14    rmarkdown_2.5    stringi_1.5.3   
[29] compiler_4.0.3   pillar_1.4.7     generics_0.1.0   httpuv_1.5.4    
[33] pkgconfig_2.0.3 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
