<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yuxin Zou" />


<title>SuSiE RSS Diagnostic using UKB</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">susierss_diagnostic</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">SuSiE RSS Diagnostic using UKB</h1>
<h4 class="author">Yuxin Zou</h4>
<h4 class="date">3/14/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-04-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>susierss_diagnostic/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20201222code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20201222)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20201222code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20201222)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomzouyuxinsusierssdiagnostictreea8c6f077f7157a85f393abea75162ad695a05c9dtargetblanka8c6f07a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/zouyuxin/susierss_diagnostic/tree/a8c6f077f7157a85f393abea75162ad695a05c9d" target="_blank">a8c6f07</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomzouyuxinsusierssdiagnostictreea8c6f077f7157a85f393abea75162ad695a05c9dtargetblanka8c6f07a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/zouyuxin/susierss_diagnostic/tree/a8c6f077f7157a85f393abea75162ad695a05c9d" target="_blank">a8c6f07</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  output/simulation_null_CLM_post.rds

Unstaged changes:
    Modified:   analysis/diagnostic_ukb_lr_null_sigma21.Rmd
    Modified:   analysis/diagnostic_ukb_lr_signal_eg.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/diagnostic_ukb_lr_signal.Rmd</code>) and HTML (<code>docs/diagnostic_ukb_lr_signal.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/zouyuxin/susierss_diagnostic/blob/a8c6f077f7157a85f393abea75162ad695a05c9d/analysis/diagnostic_ukb_lr_signal.Rmd" target="_blank">a8c6f07</a>
</td>
<td>
zouyuxin
</td>
<td>
2021-04-28
</td>
<td>
wflow_publish(“analysis/diagnostic_ukb_lr_signal.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>We want to</p>
<ol style="list-style-type: decimal">
<li><p>Check the consistency between z scores and LD matrix using <span class="math inline">\(s\)</span>;</p></li>
<li><p>Detect allele switch issue (flipped sign in z score) using conditional distribution.</p></li>
</ol>
<p>In diagnostic, we assume the following model, <span class="math display">\[
\hat{z} \sim N(0, (1-s) R + s I), 0 \leq s \leq 1.
\]</span></p>
<p>The posterior distribution of <span class="math inline">\(\hat{z}_j | \hat{z}_{-j}\)</span> is: <span class="math display">\[
\hat{z}_j | \hat{z}_{-j} \sim N(-\frac{1}{\Omega_{jj}} \Omega_{j,-j} \hat{z}_{-j},1/\Omega_{jj}), \Omega = ((1-s) R + s I)^{-1}.
\]</span></p>
<p>From the simulation, the standardized differences between observed z score and the conditional mean has longer-tailed than N(0,1). We use a mixture of normals to model the heavier tail empirically. We model the conditional distribution as <span class="math display">\[
\hat{z}_j | \hat{z}_{-j} \sim \sum_{k=1}^{K} \pi_k N(-\frac{1}{\Omega_{jj}} \Omega_{j,-j} \hat{z}_{-j},\frac{\sigma_k^2}{\Omega_{jj}}).
\]</span> The <span class="math inline">\(\sigma_k\)</span> is a fixed dense grid, the minimum value is <span class="math inline">\(0.8\)</span>, the maximum value is <span class="math inline">\(2\sqrt{max(\text{standardized difference}^2)}\)</span>. We estimate <span class="math inline">\(\pi_k\)</span> using <code>mixsqp</code>. The likelihood ratio test is based on the mixture distribution.</p>
<p>To check whether there is an allele switch issue, we test the hypothesis <span class="math inline">\(H_0:\)</span> the sign of <span class="math inline">\(\hat{z}_j\)</span> is correct, vs <span class="math inline">\(H_1:\)</span> the sign of <span class="math inline">\(\hat{z}_j\)</span> is flipped. The likelihood ratio is <span class="math display">\[
LR = \frac{p(-\hat{z}_j | \hat{z}_{-j})}{p(\hat{z}_j | \hat{z}_{-j})}
\]</span></p>
<p>There are several ways to estimate <span class="math inline">\(s\)</span>.</p>
<ol style="list-style-type: decimal">
<li><p>Let <span class="math inline">\(R = U D U^\intercal\)</span> be eigen-decomposition of <span class="math inline">\(R\)</span>, <span class="math inline">\(D\)</span> has <span class="math inline">\(p_1\)</span> positive diagonal elements, <span class="math inline">\(U = [U_1 \ U_2]\)</span>, <span class="math inline">\(U_1\)</span> is a <span class="math inline">\(p\)</span> by <span class="math inline">\(p_1\)</span> matrix, <span class="math inline">\(U_2\)</span> is a <span class="math inline">\(p\)</span> by <span class="math inline">\(p_2=p-p_1\)</span> matrix. We transform <span class="math inline">\(\hat{z}\)</span> into null space of <span class="math inline">\(R\)</span> by multiplying <span class="math inline">\(U_2^\intercal\)</span> <span class="math display">\[
U_2^T \hat{z} \sim N(0, s I_{p_2}).
\]</span> Therefore, the MLE for <span class="math inline">\(s\)</span> using partial data is <span class="math inline">\(\frac{1}{p_2} \hat{z}^\intercal U_2 U_2^\intercal \hat{z}\)</span>. If <span class="math inline">\(s &gt; 1\)</span>, we set it to 1.</p></li>
<li><p>We estimate <span class="math inline">\(s\)</span> by maximizing pseudo-likelihood, <span class="math display">\[
l(s) = \sum_{j=1}^{p} \log p(\hat{z}_j | \hat{z}_{-j}), 0 \leq s \leq 1.
\]</span></p></li>
<li><p>We estimate <span class="math inline">\(s\)</span> by maximizing <span class="math display">\[
\hat{z} \sim N(0, (1-s) R + s I), 0 \leq s \leq 1.
\]</span></p></li>
</ol>
<p><strong>Simulation Setting</strong></p>
<p>In the following simulation, we extract 200 regions from UKB. We sample 10,000 UKB individuals to simulate z scores and generate LD in. We simulate data without signal and with one signal (PVE 0.02). We sample 1,000 UKB individuals as reference panel. To get GWAS from another population, we use White blood cell count GWAS from BBJ (107,964 samples). There are 1000 SNPs in each region.</p>
<pre class="r"><code>library(tibble)
library(dplyr)
out = readRDS(&#39;data/susierss_diagnostic_ukb_query.rds&#39;)
out = as_tibble(out)

out_signal_ukb_in_1 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == FALSE, 
                                   diagnostic.z_type == &#39;ukb&#39;, diagnostic.ld_type == &#39;sample&#39;,
                                   diagnostic.method == &#39;restricted&#39;)
out_signal_ukb_in_2 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == FALSE, 
                                   diagnostic.z_type == &#39;ukb&#39;, diagnostic.ld_type == &#39;sample&#39;,
                                   diagnostic.method == &#39;pseudo&#39;)
out_signal_ukb_in_3 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == FALSE, 
                                   diagnostic.z_type == &#39;ukb&#39;, diagnostic.ld_type == &#39;sample&#39;,
                                   diagnostic.method == &#39;likelihood&#39;)

out_signal_ukb_ref_1 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == FALSE, 
                                   diagnostic.z_type == &#39;ukb&#39;, diagnostic.ld_type == &#39;ref&#39;,
                                   diagnostic.method == &#39;restricted&#39;)
out_signal_ukb_ref_2 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == FALSE, 
                                   diagnostic.z_type == &#39;ukb&#39;, diagnostic.ld_type == &#39;ref&#39;,
                                   diagnostic.method == &#39;pseudo&#39;)
out_signal_ukb_ref_3 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == FALSE, 
                                   diagnostic.z_type == &#39;ukb&#39;, diagnostic.ld_type == &#39;ref&#39;,
                                   diagnostic.method == &#39;likelihood&#39;)

out_signal_ukb_flipnull_in_1 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == TRUE, flip_z.flip_pos == &#39;null&#39;,
                                            diagnostic.z_type == &#39;ukb&#39;, diagnostic.ld_type == &#39;sample&#39;,
                                            diagnostic.method == &#39;restricted&#39;)
out_signal_ukb_flipnull_in_2 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == TRUE, flip_z.flip_pos == &#39;null&#39;,
                                            diagnostic.z_type == &#39;ukb&#39;, diagnostic.ld_type == &#39;sample&#39;,
                                            diagnostic.method == &#39;pseudo&#39;)
out_signal_ukb_flipnull_in_3 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == TRUE, flip_z.flip_pos == &#39;null&#39;,
                                            diagnostic.z_type == &#39;ukb&#39;, diagnostic.ld_type == &#39;sample&#39;,
                                            diagnostic.method == &#39;likelihood&#39;)

out_signal_ukb_flipnull_ref_1 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == TRUE, flip_z.flip_pos == &#39;null&#39;,
                                             diagnostic.z_type == &#39;ukb&#39;, diagnostic.ld_type == &#39;ref&#39;,
                                             diagnostic.method == &#39;restricted&#39;)
out_signal_ukb_flipnull_ref_2 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == TRUE, flip_z.flip_pos == &#39;null&#39;,
                                             diagnostic.z_type == &#39;ukb&#39;, diagnostic.ld_type == &#39;ref&#39;,
                                             diagnostic.method == &#39;pseudo&#39;)
out_signal_ukb_flipnull_ref_3 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == TRUE, flip_z.flip_pos == &#39;null&#39;,
                                             diagnostic.z_type == &#39;ukb&#39;, diagnostic.ld_type == &#39;ref&#39;,
                                             diagnostic.method == &#39;likelihood&#39;)

out_signal_ukb_flipsignal_in_1 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == TRUE, 
                                                flip_z.flip_pos == &#39;signal&#39;,diagnostic.z_type == &#39;ukb&#39;,
                                                diagnostic.ld_type == &#39;sample&#39;,
                                                diagnostic.method == &#39;restricted&#39;)
out_signal_ukb_flipsignal_in_2 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == TRUE, 
                                                flip_z.flip_pos == &#39;signal&#39;,diagnostic.z_type == &#39;ukb&#39;,
                                                diagnostic.ld_type == &#39;sample&#39;,
                                                diagnostic.method == &#39;pseudo&#39;)
out_signal_ukb_flipsignal_in_3 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == TRUE, 
                                                flip_z.flip_pos == &#39;signal&#39;,diagnostic.z_type == &#39;ukb&#39;,
                                                diagnostic.ld_type == &#39;sample&#39;,
                                                diagnostic.method == &#39;likelihood&#39;)

out_signal_ukb_flipsignal_ref_1 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == TRUE, 
                                               flip_z.flip_pos == &#39;signal&#39;,diagnostic.z_type == &#39;ukb&#39;,
                                               diagnostic.ld_type == &#39;ref&#39;,
                                               diagnostic.method == &#39;restricted&#39;)
out_signal_ukb_flipsignal_ref_2 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == TRUE, 
                                               flip_z.flip_pos == &#39;signal&#39;,diagnostic.z_type == &#39;ukb&#39;,
                                               diagnostic.ld_type == &#39;ref&#39;,
                                               diagnostic.method == &#39;pseudo&#39;)
out_signal_ukb_flipsignal_ref_3 = out %&gt;% filter(simulate.n_signal == 1, flip_z.flip == TRUE, 
                                               flip_z.flip_pos == &#39;signal&#39;,diagnostic.z_type == &#39;ukb&#39;,
                                               diagnostic.ld_type == &#39;ref&#39;,
                                               diagnostic.method == &#39;likelihood&#39;)

out_bbj_ref_1 = out %&gt;% filter(simulate.n_signal == 0, flip_z.flip == FALSE, flip_z.flip_pos == &#39;null&#39;,
                               diagnostic.z_type == &#39;bbj&#39;, diagnostic.ld_type == &#39;ref&#39;,
                               diagnostic.method == &#39;restricted&#39;)
out_bbj_ref_2 = out %&gt;% filter(simulate.n_signal == 0, flip_z.flip == FALSE, flip_z.flip_pos == &#39;null&#39;,
                               diagnostic.z_type == &#39;bbj&#39;, diagnostic.ld_type == &#39;ref&#39;,
                               diagnostic.method == &#39;pseudo&#39;)
out_bbj_ref_3 = out %&gt;% filter(simulate.n_signal == 0, flip_z.flip == FALSE, flip_z.flip_pos == &#39;null&#39;,
                               diagnostic.z_type == &#39;bbj&#39;, diagnostic.ld_type == &#39;ref&#39;,
                               diagnostic.method == &#39;likelihood&#39;)

out_bbj_ref_flip_1 = out %&gt;% filter(simulate.n_signal == 0, flip_z.flip == TRUE, flip_z.flip_pos == &#39;null&#39;,
                             diagnostic.z_type == &#39;bbj&#39;, diagnostic.ld_type == &#39;ref&#39;,
                             diagnostic.method == &#39;restricted&#39;)
out_bbj_ref_flip_2 = out %&gt;% filter(simulate.n_signal == 0, flip_z.flip == TRUE, flip_z.flip_pos == &#39;null&#39;,
                             diagnostic.z_type == &#39;bbj&#39;, diagnostic.ld_type == &#39;ref&#39;,
                             diagnostic.method == &#39;pseudo&#39;)
out_bbj_ref_flip_3 = out %&gt;% filter(simulate.n_signal == 0, flip_z.flip == TRUE, flip_z.flip_pos == &#39;null&#39;,
                             diagnostic.z_type == &#39;bbj&#39;, diagnostic.ld_type == &#39;ref&#39;,
                             diagnostic.method == &#39;likelihood&#39;)</code></pre>
<div id="simulation-2.1-one-signal-no-flip" class="section level2">
<h2>Simulation 2.1: one signal, no flip</h2>
<div id="check-s" class="section level3">
<h3>Check <span class="math inline">\(s\)</span></h3>
<pre class="r"><code>out_signal_ukb_in_1_s = apply(out_signal_ukb_in_1, 1, function(x) x$diagnostic.res$s)
out_signal_ukb_in_2_s = apply(out_signal_ukb_in_2, 1, function(x) x$diagnostic.res$s)
out_signal_ukb_in_3_s = apply(out_signal_ukb_in_3, 1, function(x) x$diagnostic.res$s)

out_signal_ukb_ref_1_s = apply(out_signal_ukb_ref_1, 1, function(x) x$diagnostic.res$s)
out_signal_ukb_ref_2_s = apply(out_signal_ukb_ref_2, 1, function(x) x$diagnostic.res$s)
out_signal_ukb_ref_3_s = apply(out_signal_ukb_ref_3, 1, function(x) x$diagnostic.res$s)

par(mfrow=c(1,3))
plot(out_signal_ukb_in_1_s, out_signal_ukb_ref_1_s, xlab = &#39;UKB z, LD in&#39;, ylab=&#39;UKB z, LD ref&#39;, main=&#39;MLE partial data&#39;)
abline(0,1)
plot(out_signal_ukb_in_2_s, out_signal_ukb_ref_2_s, xlab = &#39;UKB z, LD in&#39;, ylab=&#39;UKB z, LD ref&#39;, main=&#39;Pseudo likelihood&#39;)
abline(0,1)
plot(out_signal_ukb_in_3_s, out_signal_ukb_ref_3_s, xlab = &#39;UKB z, LD in&#39;, ylab=&#39;UKB z, LD ref&#39;, main=&#39;likelihood&#39;)
abline(0,1)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow=c(1,2))
plot(out_signal_ukb_ref_1_s, out_signal_ukb_ref_2_s, xlab = &#39;s from MLE partial data&#39;, ylab = &#39;s from Pseudo likelihood&#39;, 
     main = &#39;UKB z, LD ref&#39;)
abline(0,1)
plot(out_signal_ukb_ref_2_s, out_signal_ukb_ref_3_s, xlab = &#39;s from Pseudo likelihood&#39;, ylab = &#39;s from likelihood&#39;, 
     main = &#39;UKB z, LD ref&#39;)
abline(0,1)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Speed</p>
<pre class="r"><code>sp = rbind(summary(out_signal_ukb_in_1$diagnostic.DSC_TIME),
           summary(out_signal_ukb_in_2$diagnostic.DSC_TIME),
           summary(out_signal_ukb_in_3$diagnostic.DSC_TIME))
rownames(sp) = c(&#39;MLEpartial&#39;, &#39;pseudo likelihood&#39;, &#39;likelihood&#39;)
sp</code></pre>
<pre><code>                   Min. 1st Qu. Median     Mean 3rd Qu.  Max.
MLEpartial        0.931  1.0180 1.0440 1.064205 1.09100 1.450
pseudo likelihood 4.707  5.3070 5.6455 5.756775 6.28875 6.666
likelihood        1.267  1.6225 1.6770 1.680470 1.75500 2.036</code></pre>
</div>
<div id="check-conditional-mean-and-likelihood" class="section level3">
<h3>Check conditional mean and Likelihood</h3>
<pre class="r"><code>z = unlist(lapply(out_signal_ukb_in_1$diagnostic.res, function(x) x$post$z))

out_signal_ukb_in_1_z = unlist(lapply(out_signal_ukb_in_1$diagnostic.res, function(x) x$post$post_z))
out_signal_ukb_in_1_lr = unlist(lapply(out_signal_ukb_in_1$diagnostic.res, function(x) x$post$logLR))
out_signal_ukb_in_1_lrmix = unlist(lapply(out_signal_ukb_in_1$diagnostic.res, function(x) x$post$logLRmix))
out_signal_ukb_in_1_lrmix[is.infinite(out_signal_ukb_in_1_lrmix)] = -1500

out_signal_ukb_in_3_z = unlist(lapply(out_signal_ukb_in_3$diagnostic.res, function(x) x$post$post_z))
out_signal_ukb_in_3_lr = unlist(lapply(out_signal_ukb_in_3$diagnostic.res, function(x) x$post$logLR))
out_signal_ukb_in_3_lrmix = unlist(lapply(out_signal_ukb_in_3$diagnostic.res, function(x) x$post$logLRmix))
out_signal_ukb_in_3_lrmix[is.infinite(out_signal_ukb_in_3_lrmix)] = -1500

out_signal_ukb_ref_1_z = unlist(lapply(out_signal_ukb_ref_1$diagnostic.res, function(x) x$post$post_z))
out_signal_ukb_ref_1_lr = unlist(lapply(out_signal_ukb_ref_1$diagnostic.res, function(x) x$post$logLR))
out_signal_ukb_ref_1_lrmix = unlist(lapply(out_signal_ukb_ref_1$diagnostic.res, function(x) x$post$logLRmix))
out_signal_ukb_ref_1_lrmix[is.infinite(out_signal_ukb_ref_1_lrmix)] = -1500

out_signal_ukb_ref_3_z = unlist(lapply(out_signal_ukb_ref_3$diagnostic.res, function(x) x$post$post_z))
out_signal_ukb_ref_3_lr = unlist(lapply(out_signal_ukb_ref_3$diagnostic.res, function(x) x$post$logLR))
out_signal_ukb_ref_3_lrmix = unlist(lapply(out_signal_ukb_ref_3$diagnostic.res, function(x) x$post$logLRmix))
out_signal_ukb_ref_3_lrmix[is.infinite(out_signal_ukb_ref_3_lrmix)] = -1500</code></pre>
<p>We focus on those z scores with magnitude greater than 2.</p>
<pre class="r"><code>zlarge = which(abs(z) &gt; 2)
par(mfrow=c(1,2))
plot(out_signal_ukb_in_1_lrmix[zlarge], out_signal_ukb_in_3_lrmix[zlarge], xlim = c(-50,10), ylim = c(-50,10),
     xlab = &#39;logLR using s from MLE partial data&#39;, ylab = &#39;logLR using s from likelihood&#39;, 
     main = &#39;UKB z, LD in&#39;)
abline(0,1)
abline(h=0, lty=2)
abline(v=0, lty=2)

plot(out_signal_ukb_ref_1_lrmix[zlarge], out_signal_ukb_ref_3_lrmix[zlarge], xlim = c(-50,10), ylim = c(-50,10),
     xlab = &#39;logLR using s from MLE partial data&#39;, ylab = &#39;logLR using s from likelihood&#39;, 
     main = &#39;UKB z, LD ref&#39;)
abline(0,1)
abline(h=0, lty=2)
abline(v=0, lty=2)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="simulation-2.2-flipped-sign-at-signal" class="section level2">
<h2>Simulation 2.2: flipped sign at signal</h2>
<p>Based on simulation 2.1, we flipped sign of signal.</p>
<div id="check-s-1" class="section level3">
<h3>Check <span class="math inline">\(s\)</span></h3>
<pre class="r"><code>out_signal_ukb_flipsignal_in_1_s = apply(out_signal_ukb_flipsignal_in_1, 1, function(x) x$diagnostic.res$s)
out_signal_ukb_flipsignal_in_2_s = apply(out_signal_ukb_flipsignal_in_2, 1, function(x) x$diagnostic.res$s)
out_signal_ukb_flipsignal_in_3_s = apply(out_signal_ukb_flipsignal_in_3, 1, function(x) x$diagnostic.res$s)

out_signal_ukb_flipsignal_ref_1_s = apply(out_signal_ukb_flipsignal_ref_1, 1, function(x) x$diagnostic.res$s)
out_signal_ukb_flipsignal_ref_2_s = apply(out_signal_ukb_flipsignal_ref_2, 1, function(x) x$diagnostic.res$s)
out_signal_ukb_flipsignal_ref_3_s = apply(out_signal_ukb_flipsignal_ref_3, 1, function(x) x$diagnostic.res$s)

par(mfrow=c(1,3))
plot(out_signal_ukb_flipsignal_in_1_s, out_signal_ukb_flipsignal_ref_1_s, xlab = &#39;UKB z, LD in&#39;, ylab=&#39;UKB z, LD ref&#39;, main=&#39;MLE partial data&#39;)
abline(0,1)
plot(out_signal_ukb_flipsignal_in_2_s, out_signal_ukb_flipsignal_ref_2_s, xlab = &#39;UKB z, LD in&#39;, ylab=&#39;UKB z, LD ref&#39;, main=&#39;Pseudo likelihood&#39;)
abline(0,1)
plot(out_signal_ukb_flipsignal_in_3_s, out_signal_ukb_flipsignal_ref_3_s, xlab = &#39;UKB z, LD in&#39;, ylab=&#39;UKB z, LD ref&#39;, main=&#39;likelihood&#39;)
abline(0,1)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow=c(2,2))
plot(out_signal_ukb_flipsignal_in_1_s, out_signal_ukb_flipsignal_in_2_s, xlab = &#39;s from MLE partial data&#39;, ylab = &#39;s from Pseudo likelihood&#39;, 
     main = &#39;UKB z, LD in&#39;, xlim=c(0,1), ylim=c(0,1))
abline(0,1)
plot(out_signal_ukb_flipsignal_in_2_s, out_signal_ukb_flipsignal_in_3_s, xlab = &#39;s from Pseudo likelihood&#39;, ylab = &#39;s from likelihood&#39;, 
     main = &#39;UKB z, LD in&#39;, xlim=c(0,1), ylim=c(0,1))
abline(0,1)

plot(out_signal_ukb_flipsignal_ref_1_s, out_signal_ukb_flipsignal_ref_2_s, xlab = &#39;s from MLE partial data&#39;, ylab = &#39;s from Pseudo likelihood&#39;, 
     main = &#39;UKB z, LD ref&#39;, xlim=c(0,1), ylim=c(0,1))
abline(0,1)
plot(out_signal_ukb_flipsignal_ref_2_s, out_signal_ukb_flipsignal_ref_3_s, xlab = &#39;s from Pseudo likelihood&#39;, ylab = &#39;s from likelihood&#39;, 
     main = &#39;UKB z, LD ref&#39;, xlim=c(0,1), ylim=c(0,1))
abline(0,1)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="check-conditional-mean-and-likelihood-1" class="section level3">
<h3>Check conditional mean and Likelihood</h3>
<pre class="r"><code>z_flip = unlist(lapply(out_signal_ukb_flipsignal_in_1$diagnostic.res, function(x) x$post$z))

out_signal_ukb_flipsignal_in_1_z = unlist(lapply(out_signal_ukb_flipsignal_in_1$diagnostic.res, function(x) x$post$post_z))
out_signal_ukb_flipsignal_in_1_lr = unlist(lapply(out_signal_ukb_flipsignal_in_1$diagnostic.res, function(x) x$post$logLR))
out_signal_ukb_flipsignal_in_1_lrmix = unlist(lapply(out_signal_ukb_flipsignal_in_1$diagnostic.res, function(x) x$post$logLRmix))
out_signal_ukb_flipsignal_in_1_lrmix[is.infinite(out_signal_ukb_flipsignal_in_1_lrmix)] = -1500

out_signal_ukb_flipsignal_in_3_z = unlist(lapply(out_signal_ukb_flipsignal_in_3$diagnostic.res, function(x) x$post$post_z))
out_signal_ukb_flipsignal_in_3_lr = unlist(lapply(out_signal_ukb_flipsignal_in_3$diagnostic.res, function(x) x$post$logLR))
out_signal_ukb_flipsignal_in_3_lrmix = unlist(lapply(out_signal_ukb_flipsignal_in_3$diagnostic.res, function(x) x$post$logLRmix))
out_signal_ukb_flipsignal_in_3_lrmix[is.infinite(out_signal_ukb_flipsignal_in_3_lrmix)] = -1500

out_signal_ukb_flipsignal_ref_1_z = unlist(lapply(out_signal_ukb_flipsignal_ref_1$diagnostic.res, function(x) x$post$post_z))
out_signal_ukb_flipsignal_ref_1_lr = unlist(lapply(out_signal_ukb_flipsignal_ref_1$diagnostic.res, function(x) x$post$logLR))
out_signal_ukb_flipsignal_ref_1_lrmix = unlist(lapply(out_signal_ukb_flipsignal_ref_1$diagnostic.res, function(x) x$post$logLRmix))
out_signal_ukb_flipsignal_ref_1_lrmix[is.infinite(out_signal_ukb_flipsignal_ref_1_lrmix)] = -1500

out_signal_ukb_flipsignal_ref_3_z = unlist(lapply(out_signal_ukb_flipsignal_ref_3$diagnostic.res, function(x) x$post$post_z))
out_signal_ukb_flipsignal_ref_3_lr = unlist(lapply(out_signal_ukb_flipsignal_ref_3$diagnostic.res, function(x) x$post$logLR))
out_signal_ukb_flipsignal_ref_3_lrmix = unlist(lapply(out_signal_ukb_flipsignal_ref_3$diagnostic.res, function(x) x$post$logLRmix))
out_signal_ukb_flipsignal_ref_3_lrmix[is.infinite(out_signal_ukb_flipsignal_ref_3_lrmix)] = -1500</code></pre>
<pre class="r"><code>idx = which(z != z_flip)
par(mfrow=c(1,2))
plot(out_signal_ukb_flipsignal_in_1_lrmix[zlarge], out_signal_ukb_flipsignal_in_3_lrmix[zlarge], xlim = c(-20,20), ylim = c(-20,20),
     xlab = &#39;logLRmix using s from MLE partial data&#39;, ylab = &#39;logLRmix using s from likelihood&#39;, 
     main = &#39;UKB z, LD in&#39;)
points(out_signal_ukb_flipsignal_in_1_lrmix[intersect(idx, zlarge)], out_signal_ukb_flipsignal_in_3_lrmix[intersect(idx, zlarge)], col=&#39;red&#39;, pch=16)
abline(0,1)
abline(h=0, lty=2)
abline(v=0, lty=2)

plot(out_signal_ukb_flipsignal_ref_1_lrmix[zlarge], out_signal_ukb_flipsignal_ref_3_lrmix[zlarge], xlim = c(-20,15), ylim = c(-20,15),
     xlab = &#39;logLRmix using s from MLE partial data&#39;, ylab = &#39;logLRmix using s from likelihood&#39;, 
     main = &#39;UKB z, LD ref&#39;)
points(out_signal_ukb_flipsignal_ref_1_lrmix[intersect(idx, zlarge)],
       out_signal_ukb_flipsignal_ref_3_lrmix[intersect(idx, zlarge)], col=&#39;red&#39;, pch=16)
abline(0,1)
abline(h=0, lty=2)
abline(v=0, lty=2)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="simulation-2.3-flipped-sign-at-null" class="section level2">
<h2>Simulation 2.3: flipped sign at null</h2>
<p>Based on Simulation 2.1, we randomly flipped sign of one non-signal z score.</p>
<div id="check-s-2" class="section level3">
<h3>Check <span class="math inline">\(s\)</span></h3>
<pre class="r"><code>out_signal_ukb_flipnull_in_1_s = apply(out_signal_ukb_flipnull_in_1, 1, function(x) x$diagnostic.res$s)
out_signal_ukb_flipnull_in_2_s = apply(out_signal_ukb_flipnull_in_2, 1, function(x) x$diagnostic.res$s)
out_signal_ukb_flipnull_in_3_s = apply(out_signal_ukb_flipnull_in_3, 1, function(x) x$diagnostic.res$s)

out_signal_ukb_flipnull_ref_1_s = apply(out_signal_ukb_flipnull_ref_1, 1, function(x) x$diagnostic.res$s)
out_signal_ukb_flipnull_ref_2_s = apply(out_signal_ukb_flipnull_ref_2, 1, function(x) x$diagnostic.res$s)
out_signal_ukb_flipnull_ref_3_s = apply(out_signal_ukb_flipnull_ref_3, 1, function(x) x$diagnostic.res$s)

par(mfrow=c(1,3))
plot(out_signal_ukb_flipnull_in_1_s, out_signal_ukb_flipnull_ref_1_s, xlab = &#39;UKB z, LD in&#39;, ylab=&#39;UKB z, LD ref&#39;, main=&#39;MLE partial data&#39;)
abline(0,1)
plot(out_signal_ukb_flipnull_in_2_s, out_signal_ukb_flipnull_ref_2_s, xlab = &#39;UKB z, LD in&#39;, ylab=&#39;UKB z, LD ref&#39;, main=&#39;Pseudo likelihood&#39;)
abline(0,1)
plot(out_signal_ukb_flipnull_in_3_s, out_signal_ukb_flipnull_ref_3_s, xlab = &#39;UKB z, LD in&#39;, ylab=&#39;UKB z, LD ref&#39;, main=&#39;likelihood&#39;)
abline(0,1)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow=c(2,2))
plot(out_signal_ukb_flipnull_in_1_s, out_signal_ukb_flipnull_in_2_s, xlab = &#39;s from MLE partial data&#39;, ylab = &#39;s from Pseudo likelihood&#39;, 
     main = &#39;UKB z, LD in&#39;, xlim=c(0,1), ylim=c(0,1))
abline(0,1)
plot(out_signal_ukb_flipnull_in_2_s, out_signal_ukb_flipnull_in_3_s, xlab = &#39;s from Pseudo likelihood&#39;, ylab = &#39;s from likelihood&#39;, 
     main = &#39;UKB z, LD in&#39;, xlim=c(0,1), ylim=c(0,1))
abline(0,1)

plot(out_signal_ukb_flipnull_ref_1_s, out_signal_ukb_flipnull_ref_2_s, xlab = &#39;s from MLE partial data&#39;, ylab = &#39;s from Pseudo likelihood&#39;, 
     main = &#39;UKB z, LD ref&#39;, xlim=c(0,1), ylim=c(0,1))
abline(0,1)
plot(out_signal_ukb_flipnull_ref_2_s, out_signal_ukb_flipnull_ref_3_s, xlab = &#39;s from Pseudo likelihood&#39;, ylab = &#39;s from likelihood&#39;, 
     main = &#39;UKB z, LD ref&#39;, xlim=c(0,1), ylim=c(0,1))
abline(0,1)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-9-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="check-conditional-mean-and-likelihood-2" class="section level3">
<h3>Check conditional mean and Likelihood</h3>
<pre class="r"><code>z_flip = unlist(lapply(out_signal_ukb_flipnull_in_1$diagnostic.res, function(x) x$post$z))

out_signal_ukb_flipnull_in_1_z = unlist(lapply(out_signal_ukb_flipnull_in_1$diagnostic.res, function(x) x$post$post_z))
out_signal_ukb_flipnull_in_1_lr = unlist(lapply(out_signal_ukb_flipnull_in_1$diagnostic.res, function(x) x$post$logLR))
out_signal_ukb_flipnull_in_1_lrmix = unlist(lapply(out_signal_ukb_flipnull_in_1$diagnostic.res, function(x) x$post$logLRmix))
out_signal_ukb_flipnull_in_1_lrmix[is.infinite(out_signal_ukb_flipnull_in_1_lrmix)] = -1500

out_signal_ukb_flipnull_in_3_z = unlist(lapply(out_signal_ukb_flipnull_in_3$diagnostic.res, function(x) x$post$post_z))
out_signal_ukb_flipnull_in_3_lr = unlist(lapply(out_signal_ukb_flipnull_in_3$diagnostic.res, function(x) x$post$logLR))
out_signal_ukb_flipnull_in_3_lrmix = unlist(lapply(out_signal_ukb_flipnull_in_3$diagnostic.res, function(x) x$post$logLRmix))
out_signal_ukb_flipnull_in_3_lrmix[is.infinite(out_signal_ukb_flipnull_in_3_lrmix)] = -1500

out_signal_ukb_flipnull_ref_1_z = unlist(lapply(out_signal_ukb_flipnull_ref_1$diagnostic.res, function(x) x$post$post_z))
out_signal_ukb_flipnull_ref_1_lr = unlist(lapply(out_signal_ukb_flipnull_ref_1$diagnostic.res, function(x) x$post$logLR))
out_signal_ukb_flipnull_ref_1_lrmix = unlist(lapply(out_signal_ukb_flipnull_ref_1$diagnostic.res, function(x) x$post$logLRmix))
out_signal_ukb_flipnull_ref_1_lrmix[is.infinite(out_signal_ukb_flipnull_ref_1_lrmix)] = -1500

out_signal_ukb_flipnull_ref_3_z = unlist(lapply(out_signal_ukb_flipnull_ref_3$diagnostic.res, function(x) x$post$post_z))
out_signal_ukb_flipnull_ref_3_lr = unlist(lapply(out_signal_ukb_flipnull_ref_3$diagnostic.res, function(x) x$post$logLR))
out_signal_ukb_flipnull_ref_3_lrmix = unlist(lapply(out_signal_ukb_flipnull_ref_3$diagnostic.res, function(x) x$post$logLRmix))
out_signal_ukb_flipnull_ref_3_lrmix[is.infinite(out_signal_ukb_flipnull_ref_3_lrmix)] = -1500</code></pre>
<pre class="r"><code>idx = which(z != z_flip)
par(mfrow=c(1,2))
plot(out_signal_ukb_flipnull_in_1_lrmix[zlarge], out_signal_ukb_flipnull_in_3_lrmix[zlarge], xlim = c(-20,20), ylim = c(-20,20),
     xlab = &#39;logLRmix using s from MLE partial data&#39;, ylab = &#39;logLRmix using s from likelihood&#39;, 
     main = &#39;UKB z, LD in&#39;)
points(out_signal_ukb_flipnull_in_1_lrmix[intersect(idx, zlarge)], out_signal_ukb_flipnull_in_3_lrmix[intersect(idx, zlarge)], col=&#39;red&#39;, pch=16)
abline(0,1)
abline(h=0, lty=2)
abline(v=0, lty=2)

plot(out_signal_ukb_flipnull_ref_1_lrmix[zlarge], out_signal_ukb_flipnull_ref_3_lrmix[zlarge], xlim = c(-20,15), ylim = c(-20,15),
     xlab = &#39;logLRmix using s from MLE partial data&#39;, ylab = &#39;logLRmix using s from likelihood&#39;, 
     main = &#39;UKB z, LD ref&#39;)
points(out_signal_ukb_flipnull_ref_1_lrmix[intersect(idx, zlarge)],
       out_signal_ukb_flipnull_ref_3_lrmix[intersect(idx, zlarge)], col=&#39;red&#39;, pch=16)
abline(0,1)
abline(h=0, lty=2)
abline(v=0, lty=2)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="simulation-3-bbj" class="section level2">
<h2>Simulation 3: BBJ</h2>
<p>In Simulation 3.1, the z scores are from BBJ. We use reference LD matrix from UKB.</p>
<p>In simulation 3.2, we randomly flip one z score.</p>
<div id="check-s-3" class="section level3">
<h3>Check <span class="math inline">\(s\)</span></h3>
<pre class="r"><code>out_bbj_ref_1_s = apply(out_bbj_ref_1, 1, function(x) x$diagnostic.res$s)
out_bbj_ref_2_s = apply(out_bbj_ref_2, 1, function(x) x$diagnostic.res$s)
out_bbj_ref_3_s = apply(out_bbj_ref_3, 1, function(x) x$diagnostic.res$s)
out_bbj_ref_flip_1_s = apply(out_bbj_ref_flip_1, 1, function(x) x$diagnostic.res$s)
out_bbj_ref_flip_2_s = apply(out_bbj_ref_flip_2, 1, function(x) x$diagnostic.res$s)
out_bbj_ref_flip_3_s = apply(out_bbj_ref_flip_3, 1, function(x) x$diagnostic.res$s)

par(mfrow = c(1,3))
plot(out_bbj_ref_1_s, out_bbj_ref_flip_1_s, xlab = &#39;BBJ z, UKB ref LD&#39;, ylab=&#39;BBJ z, UKB ref LD, flip sign&#39;,
     main = &#39;s from MLE partial data&#39;)
abline(0,1)
plot(out_bbj_ref_2_s, out_bbj_ref_flip_2_s, xlab = &#39;BBJ z, UKB ref LD&#39;, ylab=&#39;BBJ z, UKB ref LD, flip sign&#39;,
     main = &#39;s from peudo likelihood&#39;)
abline(0,1)
plot(out_bbj_ref_3_s, out_bbj_ref_flip_3_s, xlab = &#39;BBJ z, UKB ref LD&#39;, ylab=&#39;BBJ z, UKB ref LD, flip sign&#39;,
     main = &#39;s from likelihood&#39;)
abline(0,1)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow = c(2,2))
plot(out_bbj_ref_1_s, out_bbj_ref_2_s, main = &#39;BBJ z, UKB ref LD&#39;, 
     xlab = &#39;s from MLE partial data&#39;, ylab=&#39;s from peudo likelihood&#39;)
abline(0,1)
plot(out_bbj_ref_2_s, out_bbj_ref_3_s, main = &#39;BBJ z, UKB ref LD&#39;, 
     xlab = &#39;s from peudo likelihood&#39;, ylab=&#39;s from likelihood&#39;)
abline(0,1)

plot(out_bbj_ref_flip_1_s, out_bbj_ref_flip_2_s, main = &#39;BBJ z, UKB ref LD, flip sign&#39;, 
     xlab = &#39;s from MLE partial data&#39;, ylab=&#39;s from peudo likelihood&#39;)
abline(0,1)
plot(out_bbj_ref_flip_2_s, out_bbj_ref_flip_3_s, main = &#39;BBJ z, UKB ref LD, flip sign&#39;, 
     xlab = &#39;s from peudo likelihood&#39;, ylab=&#39;s from likelihood&#39;)
abline(0,1)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="check-conditional-mean-and-likelihood-3" class="section level3">
<h3>Check conditional mean and Likelihood</h3>
<pre class="r"><code>z = unlist(lapply(out_bbj_ref_1$diagnostic.res, function(x) x$post$z))

out_bbj_ref_1_z = unlist(lapply(out_bbj_ref_1$diagnostic.res, function(x) x$post$post_z))
out_bbj_ref_1_lr = unlist(lapply(out_bbj_ref_1$diagnostic.res, function(x) x$post$logLR))
out_bbj_ref_1_lrmix = unlist(lapply(out_bbj_ref_1$diagnostic.res, function(x) x$post$logLRmix))

out_bbj_ref_3_z = unlist(lapply(out_bbj_ref_3$diagnostic.res, function(x) x$post$post_z))
out_bbj_ref_3_lr = unlist(lapply(out_bbj_ref_3$diagnostic.res, function(x) x$post$logLR))
out_bbj_ref_3_lrmix = unlist(lapply(out_bbj_ref_3$diagnostic.res, function(x) x$post$logLRmix))

z_flip = unlist(lapply(out_bbj_ref_flip_1$diagnostic.res, function(x) x$post$z))

out_bbj_ref_flip_1_z = unlist(lapply(out_bbj_ref_flip_1$diagnostic.res, function(x) x$post$post_z))
out_bbj_ref_flip_1_lr = unlist(lapply(out_bbj_ref_flip_1$diagnostic.res, function(x) x$post$logLR))
out_bbj_ref_flip_1_lrmix = unlist(lapply(out_bbj_ref_flip_1$diagnostic.res, function(x) x$post$logLRmix))

out_bbj_ref_flip_3_z = unlist(lapply(out_bbj_ref_flip_3$diagnostic.res, function(x) x$post$post_z))
out_bbj_ref_flip_3_lr = unlist(lapply(out_bbj_ref_flip_3$diagnostic.res, function(x) x$post$logLR))
out_bbj_ref_flip_3_lrmix = unlist(lapply(out_bbj_ref_flip_3$diagnostic.res, function(x) x$post$logLRmix))</code></pre>
<pre class="r"><code>zlarge = which(abs(z) &gt; 2)
idx = which(z != z_flip)
par(mfrow=c(1,2))
plot(out_bbj_ref_1_lrmix[zlarge], out_bbj_ref_3_lrmix[zlarge], xlim = c(-20,20), ylim = c(-20,20),
     xlab = &#39;logLRmix using s from MLE partial data&#39;, ylab = &#39;logLRmix using s from likelihood&#39;, 
     main = &#39;UKB z, LD in&#39;)
abline(0,1)
abline(h=0, lty=2)
abline(v=0, lty=2)

plot(out_bbj_ref_flip_1_lrmix[zlarge], out_bbj_ref_flip_3_lrmix[zlarge], xlim = c(-20,20), ylim = c(-20,20),
     xlab = &#39;logLRmix using s from MLE partial data&#39;, ylab = &#39;logLRmix using s from likelihood&#39;, 
     main = &#39;UKB z, LD in&#39;)
points(out_bbj_ref_flip_1_lrmix[intersect(idx, zlarge)],
       out_bbj_ref_flip_3_lrmix[intersect(idx, zlarge)], col=&#39;red&#39;, pch=16)
abline(0,1)
abline(h=0, lty=2)
abline(v=0, lty=2)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Comparing LR</p>
<pre class="r"><code>plot(out_bbj_ref_1_lrmix[zlarge], out_bbj_ref_flip_1_lrmix[zlarge], xlim = c(-50,20), ylim=c(-50,20),
     xlab=&#39;logLRmix No flip&#39;,
     ylab=&#39;logLRmix flip&#39;, main=&#39;BBJ z, UKB ref LD, s from MLE partial data&#39;)
points(out_bbj_ref_1_lrmix[intersect(idx, zlarge)], out_bbj_ref_flip_1_lrmix[intersect(idx, zlarge)], col=&#39;red&#39;, pch=16)
abline(0,1)
abline(h=0, v=0, lty=4)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(out_bbj_ref_3_lrmix[zlarge], out_bbj_ref_flip_3_lrmix[zlarge], xlim = c(-50,20), ylim=c(-50,20),
     xlab=&#39;logLRmix No flip&#39;,
     ylab=&#39;logLRmix flip&#39;, main=&#39;BBJ z, UKB ref LD, s from likelihood&#39;)
points(out_bbj_ref_3_lrmix[intersect(idx, zlarge)], out_bbj_ref_flip_3_lrmix[intersect(idx, zlarge)], col=&#39;red&#39;, pch=16)
abline(0,1)
abline(h=0, v=0, lty=4)</code></pre>
<p><img src="figure/diagnostic_ukb_lr_signal.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.0.5     tibble_3.1.1    workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6        pillar_1.6.0      compiler_4.0.3    later_1.1.0.1    
 [5] git2r_0.28.0      highr_0.8         tools_4.0.3       digest_0.6.27    
 [9] evaluate_0.14     lifecycle_1.0.0   pkgconfig_2.0.3   rlang_0.4.10     
[13] DBI_1.1.1         yaml_2.2.1        xfun_0.22         stringr_1.4.0    
[17] knitr_1.31        generics_0.1.0    fs_1.5.0          vctrs_0.3.7      
[21] rprojroot_2.0.2   tidyselect_1.1.0  glue_1.4.2        R6_2.5.0         
[25] fansi_0.4.2       rmarkdown_2.7     purrr_0.3.4       magrittr_2.0.1   
[29] whisker_0.4       promises_1.2.0.1  ellipsis_0.3.1    htmltools_0.5.1.1
[33] assertthat_0.2.1  httpuv_1.5.5      utf8_1.2.1        stringi_1.5.3    
[37] crayon_1.4.1     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
