---
title: "SuSiE RSS UKB Diagnostic Example (flip at null)"
author: "Yuxin Zou"
date: "3/14/2021"
output: 
  workflowr::wflow_html:
    code_folding: hide
---

There are 1002 SNPs in the region.
```{r}
library(mixsqp)
dat = readRDS('data/rss_diagnostic_ukb_flipnull_eg.rds')
z = dat$z
zorig = dat$z_orig
ld_sample = dat$ld[['sample']]
ld_ref = dat$ld[['ref']]
idx = which(dat$true_coef!=0)
idx_flip = which(z != zorig)
```

The original z scores are
```{r}
# pdf('~/Documents/GitHub/Yuxin_Thesis/susie_rss/figures/alleleflip.pdf', width =15, height = 5, pointsize=24)
par(mfrow=c(1,3), cex.axis = 0.9)
plot(zorig, pch = 16, col = '#767676', main = 'Marginal Associations', 
     xlab='SNP', ylab = 'z scores')
points(idx, zorig[idx], col=2, pch=16)
points(idx_flip, zorig[idx_flip], col=7, pch=16)
f1 = susieR::susie_rss(z, ld_ref)
susieR::susie_plot(f1, y='PIP', b = dat$true_coef, main = 'Result with allele mis-match',
                   cex.main=0.9)
points(idx_flip, f1$pip[idx_flip], col=7, pch=16, cex = 0.8)
f2 = susieR::susie_rss(zorig, ld_ref)
susieR::susie_plot(f2, y='PIP', b = dat$true_coef, main = 'Result with correct allele alignment',
                   cex.main=0.9)
points(idx_flip, f2$pip[idx_flip], col=7, pch=16)
# dev.off()
```


```{r}
posterior_cond = function(z, precision){
  postmean = c()
  postvar = c()
  for(i in 1:length(z)){
    postmean = c(postmean, -(1/precision[i,i]) * precision[i,-i] %*% z[-i])
    postvar = c(postvar, 1/precision[i,i])
  }
  prob = pnorm(z, postmean, sd = sqrt(postvar))
  prob = pmin(prob, 1-prob)*2
  logl0 = dnorm(z, postmean, sd = sqrt(postvar), log = T)
  logl1 = dnorm(-z, postmean, sd = sqrt(postvar), log = T)
  logLR = logl1 - logl0
  post_z = (z-postmean)/sqrt(postvar)
  
  a_min = 0.8
  if(max(post_z^2) < 1){
    a_max = 2
  }else{
    a_max = 2*sqrt(max(post_z^2))
  }
  npoint = ceiling(log2(a_max/a_min)/log2(1.05))
  a_grid = 1.05^((-npoint):0) * a_max
  sd_mtx = outer(sqrt(postvar), a_grid)
  matrix_llik = dnorm(z - postmean, sd=sd_mtx, log=T)
  lfactors    <- apply(matrix_llik,1,max)
  matrix_llik <- matrix_llik - lfactors
  w = mixsqp(matrix_llik, log=T, control = list(verbose=FALSE))$x
  
  logl0mix = as.numeric(log(exp(matrix_llik) %*% w)) + lfactors
  matrix_llik = dnorm(z + postmean, sd=sd_mtx, log=T)
  lfactors    <- apply(matrix_llik,1,max)
  matrix_llik <- matrix_llik - lfactors
  logl1mix = as.numeric(log(exp(matrix_llik) %*% w)) + lfactors
  logLRmix = logl1mix - logl0mix
  postvar_mix = (sd_mtx^2) %*% w
  
  return(list(z = z, postmean = postmean, postvar = postvar, post_z = post_z, 
                    logl0 = logl0, logl1 = logl1, logLR = logLR, 
                    w = w, postvarmix = postvar_mix,
                    logl0mix = logl0mix, logl1mix = logl1mix, logLRmix = logLRmix))
}

```

We flip the green dot.
```{r}
eigenld = eigen(ld_ref)
eigenld$values[eigenld$values < 1e-8] = 0
negloglikelihood = function(s, z, eigenld){
    0.5 * sum(log((1-s)*eigenld$values+s)) + 
      0.5 * sum(z * eigenld$vectors %*% ((t(eigenld$vectors) * (1/((1-s)*eigenld$values + s))) %*% z))
}
s = optim(0.5, fn=negloglikelihood,z=z, eigenld=eigenld,
          method = 'Brent', lower=0, upper=1)$par
s
precision = eigenld$vectors %*% (t(eigenld$vectors) * (1/((1-s)*eigenld$values + s)))
res = list(s = s, post = posterior_cond(z, precision))
```




